pandas/core/frame.py

```diff
@@ -1274,16 +1275,31 @@ def from_items(cls, items, columns=None, orient='columns'):
                 columns = _ensure_index(keys)
                 arrays = values
 
-            return cls._from_arrays(arrays, columns, None)
+            try:
+                return cls._from_arrays(arrays, columns, None)
+
+            except ValueError:
```

Comment:
why are you not simply doing this *inside* ``._from_arrays`` ? you only would need this once

Reply:
In the case where ``orient == 'index'`` we don't get to ``._from_arrays`` because the previous line 
```
data = [lib.maybe_convert_objects(v) for v in arr]
```
throws the error:
```
TypeError: Argument 'objects' has incorrect type (expected numpy.ndarray, got int)
```

Reply:
ok i see what you are doing. can you add a comment to each section about what you are guarding against here.
we normally don't like to try/except around multiple statements but this is really a 'more informative message' guard.